### ISSUE TITLE
IsarError: SchemaError: Target collection does not exist.

### ISSUE THREAD

**Issue by CalsRanna** [isar/isar#1149]
### Steps to Reproduce

When I migrate a schema from version before to latest auto, the app can not work and give me an error like 'IsarError: SchemaError: Target collection does not exist.'

What I did was put some code into embedded from in the schema code.

### Code sample

```dart
import 'package:isar/isar.dart';

part 'source.g.dart';

@collection
@Name('sources')
class Source {
  Id id = Isar.autoIncrement;
  String? name;
  String? url;
  bool enabled = true;
  @Name('explore_enabled')
  bool exploreEnabled = false;
  String type = 'book';
  String? group;
  String? comment;
  @Name('login_url')
  String? loginUrl;
  @Name('url_regex')
  String? urlRegex;
  String? header;
  String charset = 'utf8';
  short? weight;
  short order = 0;
  @Name('search_url')
  String? searchUrl;
  @Name('search_check_credential')
  String? searchCheckCredential;
  @Name('search_books')
  String? searchBooks;
  @Name('search_name')
  String? searchName;
  @Name('search_author')
  String? searchAuthor;
  @Name('search_category')
  String? searchCategory;
  @Name('search_word_count')
  String? searchWordCount;
  @Name('search_introduction')
  String? searchIntroduction;
  @Name('search_cover')
  String? searchCover;
  @Name('search_information_url')
  String? searchInformationUrl;
  @Name('search_latest_chapter')
  String? searchLatestChapter;
  @Name('information_preprocess')
  String? informationPreprocess;
  @Name('information_name')
  String? informationName;
  @Name('information_author')
  String? informationAuthor;
  @Name('information_category')
  String? informationCategory;
  @Name('information_word_count')
  String? informationWordCount;
  @Name('information_latest_chapter')
  String? informationLatestChapter;
  @Name('information_introduction')
  String? informationIntroduction;
  @Name('information_cover')
  String? informationCover;
  @Name('information_catalogue_url')
  String? informationCatalogueUrl;
  EmbeddedCatalogue? catalogue;
  @Name('content_content')
  String? contentContent;
  @Name('content_pagination')
  String? contentPagination;
  @Name('content_replace')
  String? contentReplace;
  @Name('explore_url')
  String? exploreUrl;
  @Name('explore_books')
  String? exploreBooks;
  @Name('explore_name')
  String? exploreName;
  @Name('explore_author')
  String? exploreAuthor;
  @Name('explore_category')
  String? exploreCategory;
  @Name('explore_word_count')
  String? exploreWordCount;
  @Name('explore_latest_chapter')
  String? exploreLatestChapter;
  @Name('explore_introduction')
  String? exploreIntroduction;
  @Name('explore_cover')
  String? exploreCover;
  @Name('explore_information_url')
  String? exploreInformationUrl;
}

@embedded
@Name('embedded_catalogues')
class EmbeddedCatalogue {
  String? chapters;
  String? name;
  String? url;
  String? vip;
  @Name('updated_at')
  String? updatedAt;
  String? pagination;
}
```

### Details

 - Platform: iPhone 14 Pro Max
 - Flutter version: 3.7.7
 - Isar version: 3.0.5

---

**Comment by thanhnamitit** [comment]
I meet the same issue :((. Poor me

**Comment by celvineadiputra-dev** [comment]
Try this sir, remove this `@Name('embedded_catalogues')`, and run again ` flutter pub run build_runner build`
it's work for me 😀

Thank you

**Comment by thanhnamitit** [comment]
> Try this sir, remove this `@Name('embedded_catalogues')`, and run again ` flutter pub run build_runner build` it's work for me 😀
> 
> Thank you

Thank you for your input. In short, it is not necessary to assign a name to an embedded object.

**Comment by simc** [comment]
This is fixed in v3.1.0 which will be available soon on pub.dev.
